### ISSUE TITLE
raw noise vectors are not available in `datatree`

### ISSUE THREAD

**Issue by agrouaze** [umr-lops/xsar#140]
**What is given in annotations of S1 products:**

```
'noise.range.line', 'noise.range.sample', 'noise.range.noiseLut'
```
and
```
    'noise.azi.line', 'noise.azi.line_start', 'noise.azi.line_stop',
            'noise.azi.sample_start',
            'noise.azi.sample_stop', 'noise.azi.noiseLut',
            'noise.azi.swath')
```
it looks like :
![image](https://user-images.githubusercontent.com/14925067/220384226-3e9013fd-08a5-4ab1-8b20-9eedfc413d33.png)

note: 

1. for IW before IPF2.8 the noise what only stored are range profiles, after it is range and azimuth.
2. for IW before 2017 the noise range vectors didn't have the same size
3. for WV there is no noise vector in azimuth

**What is currently done in `xsar` to get final NESZ matrix:**
1) defines the images blocks (ie tiles) where noise range and noise azi are intersecting, to have an inteprolation function per block . see [this part of the code](https://github.com/umr-lops/xsar/blob/0620dcf49de9f27fa996152cc38ba592681d33d4/src/xsar/sentinel1_xml_mappings.py#L281
2) defines the intersection between the subset asked by user and the noise blocks . see class [_NoiseLut](https://github.com/umr-lops/xsar/blob/0620dcf49de9f27fa996152cc38ba592681d33d4/src/xsar/sentinel1_xml_mappings.py#L243) 
3) apply patch on noise vector for WV IPF 2.9 and 2.91.  : https://github.com/umr-lops/xsar/blob/905c367538bd6e698a7168d5c5119c37ce1d7dc9/src/xsar/sentinel1_dataset.py#L673
3) multiply noise_azi and noise_range to get uncalibrated noise (see https://github.com/umr-lops/xsar/blob/0620dcf49de9f27fa996152cc38ba592681d33d4/src/xsar/sentinel1_dataset.py#L365) 
4) calibrate the noise to get NESZ doing `NESZ = noise_values / sigma0_lut² ` see : https://github.com/umr-lops/xsar/blob/0620dcf49de9f27fa996152cc38ba592681d33d4/src/xsar/sentinel1_dataset.py#L1589 

**What can be given to SLC users:**

for noise in azimuth
![image](https://user-images.githubusercontent.com/14925067/220408862-81d089d6-f7df-435d-ade4-08b3c67f1966.png)
for noise in range
![image](https://user-images.githubusercontent.com/14925067/220414349-05b87c94-f633-4f02-a43a-6b1fe6b3e9f1.png)

**Comment by agrouaze** [comment]
so far no official paper from MPC group regarding denoising have been published.

**Comment by agrouaze** [comment]
fixed in #141
